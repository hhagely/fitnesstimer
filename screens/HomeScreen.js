import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import { WebBrowser } from 'expo';
import AmrapTimer from '../components/timers/Amrap';

export default class HomeScreen extends React.Component {
	constructor(props) {
		super(props);

		this.state = this.initialState = {
			isRunning: false,
			timeElapsed: 0,
			countdownTimeLeft: 0,
			startTime: 0,
			timerSettings: {
				timerType: 'AMRAP',
				timerDuration: 10,
				countdown: true,
				countdownDuration: 10
			}
		};
	}

	static navigationOptions = {
		header: null
	};

	renderTimer() {
		console.log('calling render timer');
		const timerSettings = this.props.navigation.getParam(
			'timerSettings',
			this.initialState.timerSettings
		);

		// TODO: should not be passing identical props into 2 separate components like below. Separate it out.
		switch (timerSettings.timerType) {
			case 'AMRAP':
				return <AmrapTimer timerSettings={timerSettings} />;
				break;
			// case 'Emom':
			// 	return (
			// 		<EmomTimer
			// 			render={(timer) => (
			// 				<TimeElapsed
			// 					timeElapsed={timeElapsed}
			// 					isRunning={isRunning}
			// 					timerSettings={timerSettings}
			// 				/>
			// 			)}
			// 		/>
			// 	);
			// 	break;
			// case 'Tabata':
			// 	return <TabataTimer duration={timerSettings.timerDuration} />;
			// 	break;
			// case 'ReverseTabata':
			// 	return <ReverseTabataTimer duation={timerSettings.timerDuration} />;
			// 	break;
			default:
				return <View />;
		}
	}

	render() {
		return (
			<View style={styles.container}>
				<ScrollView contentContainerStyle={styles.contentContainer}>
					<View style={styles.timerContainer}>{this.renderTimer()}</View>
				</ScrollView>
			</View>
		);
	}

	_maybeRenderDevelopmentModeWarning() {
		if (__DEV__) {
			const learnMoreButton = (
				<Text onPress={this._handleLearnMorePress} style={styles.helpLinkText}>
					Learn more
				</Text>
			);

			return (
				<Text style={styles.developmentModeText}>
					Development mode is enabled, your app will be slower but you can use
					useful development tools. {learnMoreButton}
				</Text>
			);
		} else {
			return (
				<Text style={styles.developmentModeText}>
					You are not in development mode, your app will run at full speed.
				</Text>
			);
		}
	}

	_handleLearnMorePress = () => {
		WebBrowser.openBrowserAsync(
			'https://docs.expo.io/versions/latest/guides/development-mode'
		);
	};

	_handleHelpPress = () => {
		WebBrowser.openBrowserAsync(
			'https://docs.expo.io/versions/latest/guides/up-and-running.html#can-t-see-your-changes'
		);
	};
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff'
	},
	timerContainer: {
		flex: 1,
		backgroundColor: '#fff',
		justifyContent: 'space-between'
	},
	developmentModeText: {
		marginBottom: 20,
		color: 'rgba(0,0,0,0.4)',
		fontSize: 14,
		lineHeight: 19,
		textAlign: 'center'
	},
	contentContainer: {
		flex: 1,
		paddingTop: 30
	}
});
