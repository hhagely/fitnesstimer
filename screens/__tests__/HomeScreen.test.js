import 'react-native';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';

import { HomeScreen } from '../HomeScreen';

configure({ adapter: new Adapter() });

describe('<HomeScreen />', () => {
	it('Renders the Homescreen component', () => {
		const tree = shallow(<HomeScreen />);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
