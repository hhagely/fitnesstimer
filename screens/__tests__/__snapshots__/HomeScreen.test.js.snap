// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> Renders the Homescreen component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "space-between",
        }
      }
    />
  </ScrollView>
</View>
`;
